src/ansys/mapdl/core/launcher.py

```diff
@@ -1750,11 +1750,25 @@ def launch_mapdl(
         check_valid_port(port)
         LOG.debug(f"Using default port {port}")
 
+    # verify version
+    if exec_file and version:
+        raise ValueError("Cannot specify both ``exec_file`` and ``version``.")
+
+    if version is None:
+        version = os.getenv("PYMAPDL_MAPDL_VERSION", None)
+
+    version = _verify_version(version)  # return a int version or none
+
     # Start MAPDL with PyPIM if the environment is configured for it
     # and the user did not pass a directive on how to launch it.
-    if _HAS_PIM and exec_file is None and pypim.is_configured():
+    if _HAS_PIM and exec_file is None and pypim.is_configured():  # pragma: no cover
```

Comment:
oh are you sure about the `no cover`? I thought this part was covered

Reply:
You are right!